packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -247,55 +280,90 @@ class IndexedDbRemoteDocumentCacheImpl implements IndexedDbRemoteDocumentCache {
       });
   }
 
-  getAll(
+  getAllFromCollection(
     transaction: PersistenceTransaction,
     collection: ResourcePath,
-    sinceReadTime: SnapshotVersion
+    offset: IndexOffset
   ): PersistencePromise<MutableDocumentMap> {
-    let results = mutableDocumentMap();
-
-    const immediateChildrenPathLength = collection.length + 1;
-
-    const iterationOptions: IterateOptions = {};
-    if (sinceReadTime.isEqual(SnapshotVersion.min())) {
-      // Documents are ordered by key, so we can use a prefix scan to narrow
-      // down the documents we need to match the query against.
-      const startKey = collection.toArray();
-      iterationOptions.range = IDBKeyRange.lowerBound(startKey);
-    } else {
-      // Execute an index-free query and filter by read time. This is safe
-      // since all document changes to queries that have a
-      // lastLimboFreeSnapshotVersion (`sinceReadTime`) have a read time set.
-      const collectionKey = collection.toArray();
-      const readTimeKey = toDbTimestampKey(sinceReadTime);
-      iterationOptions.range = IDBKeyRange.lowerBound(
-        [collectionKey, readTimeKey],
-        /* open= */ true
-      );
-      iterationOptions.index = DbRemoteDocumentCollectionReadTimeIndex;
-    }
+    const startKey = [
+      collection.popLast().toArray(),
```

Comment:
Can we find a way to make it more readable, like signalling this is a `collectionPrefix` and below is `collectionId` somehow?

Reply:
Added comments to the helpers.